### PR TITLE
Add update and terminate declaration functions to rpc

### PR DESCRIPTION
**Pull request by progsmile**
edenlabllc/ehealth.api#4092

### CODE REVIEW

File: apps/core/lib/core/rpc/rpc.ex [hunk 1]
```diff
@@ -142,4 +143,28 @@ defmodule Core.Rpc do
...
+
+  @doc """
+  Update declaration
+
+  ## Examples
+      iex> Core.Rpc.update_declaration(%Core.Declarations.Declaration{}, %{"status" => "active"})
+      {:ok, %Core.Declarations.Declaration{}}
+  """
+  @spec update_declaration(%Core.Declarations.Declaration{}, map) :: {:ok, %Core.Declarations.Declaration{}}
+  def update_declaration(%Declaration{} = declaration, %{} = params) do
```
Comment (progsmile) [inline apps/core/lib/core/rpc/rpc.ex]: What do you think about passing previously received declaration struct?

Comment (progsmile) [reply apps/core/lib/core/rpc/rpc.ex]: Suppose it would be flexible to use `id`, but would do extra db request

Comment (PavelVesnin) [reply apps/core/lib/core/rpc/rpc.ex]: so if you have just `id`, you need to pass Schema anyway?

File: apps/core/test/unit/rpc_test.exs [hunk 2]
```diff
@@ -205,4 +207,40 @@ defmodule Core.RpcTest do
...
+
+    test "invalid status transaction" do
+      declaration = insert(:declaration, status: "closed")
+      patch = %{status: "closed", updated_by: UUID.generate()}
+
+      assert {:error, %Ecto.Changeset{valid?: false}} = Rpc.update_declaration(declaration.id, patch)
+    end
+
+    test "not found" do
+      assert nil == Rpc.update_declaration(UUID.generate(), %{})
```
Comment (PavelVesnin) [inline apps/core/test/unit/rpc_test.exs]:
```suggestion
      refute Rpc.update_declaration(UUID.generate(), %{})
```

File: apps/core/lib/core/rpc/rpc.ex [hunk 3]
```diff
@@ -142,4 +143,30 @@ defmodule Core.Rpc do
...
+
+  @doc """
+  Update declaration
+
+  ## Examples
+      iex> Core.Rpc.update_declaration("0042500e-6ac0-45fb-b82a-25f7857c49a8", %{"status" => "active"})
+      {:ok, %Core.Declarations.Declaration{}}
+  """
+  @spec update_declaration(binary, map) :: {:ok, %Core.Declarations.Declaration{}} | nil | {:error, %Ecto.Changeset{}}
+  def update_declaration(id, %{} = params) do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/rpc/rpc.ex]: which params are allowed?

File: apps/core/lib/core/rpc/rpc.ex [hunk 4]
```diff
@@ -142,4 +143,30 @@ defmodule Core.Rpc do
...
+
+  @doc """
+  Terminate declaration
+
+  ## Examples
+      iex> Core.Rpc.terminate_declaration("0042500e-6ac0-45fb-b82a-25f7857c49a8", %{"updated_by" => "11225aae-7ac0-45fb-b82a-25f7857c49b0"})
+      {:ok, %Core.Declarations.Declaration{}}
+  """
+  @spec terminate_declaration(binary, map) :: {:ok, %Core.Declarations.Declaration{}} | %Ecto.Changeset{}
+  def terminate_declaration(id, %{"updated_by" => _} = params) do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/rpc/rpc.ex]: which params are allowed?

File: apps/core/lib/core/rpc/rpc.ex (hunk 3 above)
Comment (progsmile) [reply apps/core/lib/core/rpc/rpc.ex]: All from `declaration_changeset` function, wanna spec them all?
image

Comment (AlexKovalevych) [reply apps/core/lib/core/rpc/rpc.ex]: yes, it should be reflected in the doc, since there is no other doc for rpc functions

**Pull request merged**